Gems/Terrain/Code/Source/Components/TerrainWorldRendererComponent.cpp

```diff
@@ -100,52 +93,20 @@ namespace Terrain
-                        ->Attribute(AZ::Edit::Attributes::SoftMin, 512.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 16.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 128.0f)
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapMaxResolution,
+                        "Detail clipmap max resolution (in texels per meter)",
+                        "The resolution of the highest resolution clipmap in the stack.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.5f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 256.0f)
                         ->Attribute(AZ::Edit::Attributes::Max, 2048.0f)
-                        ->Attribute(AZ::Edit::Attributes::SoftMax, 2048.0f)
-                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_macroClipmapScaleBase, "Macro clipmap scale base", "The scale base between two adjacent clipmap layers.")
-                        ->Attribute(AZ::Edit::Attributes::Min, 2.0f)
-                        ->Attribute(AZ::Edit::Attributes::SoftMin, 2.0f)
-                        ->Attribute(AZ::Edit::Attributes::Max, 4.0f)
-                        ->Attribute(AZ::Edit::Attributes::SoftMax, 4.0f)
-                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapScaleBase, "Detail clipmap scale base", "The scale base between two adjacent clipmap layers.")
-                        ->Attribute(AZ::Edit::Attributes::Min, 2.0f)
-                        ->Attribute(AZ::Edit::Attributes::SoftMin, 2.0f)
-                        ->Attribute(AZ::Edit::Attributes::Max, 4.0f)
-                        ->Attribute(AZ::Edit::Attributes::SoftMax, 4.0f)
-                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_macroClipmapMarginSize, "Macro clipmap margin size", "The margin of the clipmap where the data won't be used.")
-                        ->Attribute(AZ::Edit::Attributes::Min, 1u)
-                        ->Attribute(AZ::Edit::Attributes::SoftMin, 1u)
-                        ->Attribute(AZ::Edit::Attributes::Max, 8u)
-                        ->Attribute(AZ::Edit::Attributes::SoftMax, 8u)
-                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapMarginSize, "Detail clipmap margin size", "The margin of the clipmap where the data won't be used.")
-                        ->Attribute(AZ::Edit::Attributes::Min, 1u)
-                        ->Attribute(AZ::Edit::Attributes::SoftMin, 1u)
-                        ->Attribute(AZ::Edit::Attributes::Max, 8u)
-                        ->Attribute(AZ::Edit::Attributes::SoftMax, 8u)
-                    // Note: m_extendedClipmapMarginSize, m_clipmapBlendSize won't be exposed because algorithm may change and we may not need them.
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 4096.0f)
```

Comment:
So I was originally imagining that the macro material that would be composited into the detail materials would just be the first macro material clipmap, whose size would just be derived based on when the detail material rendering distance. However I think this is a little better because it allows for higher detail macro material clipmaps that are never used directly, but would be used when compositing the detail clipmaps. (correct me if I don't understand something about this)

This still allows for some weird ux issues that I'm not sure how to solve - like what if someone made their macro material resolution _higher_ than the detail material resolution? I don't think it would make sense to have a macro material that was greater than half the resolution of the detail material, and generally it'd be much much less (like 1/256).